crates/context_aware_config/src/api/context/helpers.rs

```diff
@@ -17,16 +17,16 @@ use diesel::{
     r2d2::{ConnectionManager, PooledConnection},
     ExpressionMethods, PgConnection, QueryDsl, RunQueryDsl,
 };
-use serde_json::{Map, Value};
+use serde_json::{json, Map, Value};
 use std::collections::HashMap;
 type DBConnection = PooledConnection<ConnectionManager<PgConnection>>;
 
 pub fn validate_condition_with_functions(
     conn: &mut DBConnection,
-    context: &Value,
+    condition: &Condition,
 ) -> superposition::Result<()> {
     use dimensions::dsl;
-    let context = extract_dimensions(context)?;
+    let context = extract_dimensions(&json!(**condition))?;
```

Comment:
derive_more has an auto deref trait. Can we use that to avoid our code looking like C++?

Reply:
Can we also add a to_json() method on newtypes? Skip one more step here